Traceback (most recent call last):
  File "/home/sean/anaconda3/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/home/sean/anaconda3/lib/python3.8/site-packages/nbclient/client.py", line 1112, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/home/sean/anaconda3/lib/python3.8/site-packages/nbclient/util.py", line 74, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/home/sean/anaconda3/lib/python3.8/site-packages/nbclient/util.py", line 53, in just_run
    return loop.run_until_complete(coro)
  File "/home/sean/anaconda3/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/home/sean/anaconda3/lib/python3.8/site-packages/nbclient/client.py", line 553, in async_execute
    await self.async_execute_cell(
  File "/home/sean/anaconda3/lib/python3.8/site-packages/nbclient/client.py", line 857, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/home/sean/anaconda3/lib/python3.8/site-packages/nbclient/client.py", line 760, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
def p3(x, a_0, a_1, a_2, a_3):
    return a_0 + a_1*x + a_2*x**2 + a_3*x**3 

def p1(x, a_0, a_1):
    return a_0 + a_1*x
# Fit Coefs
##Unweighted Least Squares Global
p1_global_fit, p1_global_error = scipy.optimize.curve_fit(p1,
    co2_data_global['decimal'],
    co2_data_global['average']
    )
#chisq
y_p1_global = p1(co2_data_global['decimal'], *p1_global_fit)
chisq_global_p1 = np.sum(((y_p1_global-co2_data_global['decimal'])/co2_data_global['decimal'])**2)
#p3
p3_global_fit, p3_global_error = scipy.optimize.curve_fit(p3,
    co2_data_global['decimal'],
    co2_data_global['average']
    )
#chisq
y_p3_global = p3(co2_data_global['decimal'], *p3_global_fit)
chisq_global_p3 = np.sum(((y_p3_global-co2_data_global['decimal'])/co2_data_global['decimal'])**2)
# Weighted Least Squares Mauna Loa 
p1_ml_fit, p1_ml_error = scipy.optimize.curve_fit(p1,
    co2_data_ml['decimal date'],
    co2_data_ml['average'],
    sigma = co2_data_ml['sdev']
    )
y_p1_ml = p3(co2_data_ml['decimal date'], *p3_ml_fit)
chisq_ml_p1 = np.sum(((y_p1_ml-co2_data_ml['decimal date'])/co2_data_ml['sdev'])**2)

p3_ml_fit, p3_ml_error = scipy.optimize.curve_fit(p3,
    co2_data_ml['decimal date'],
    co2_data_ml['average'],
    sigma = co2_data_ml['sdev']
    )
y_p3_ml = p3(co2_data_ml['decimal date'], *p3_ml_fit)
chisq_ml_p3 = np.sum(((y_p3_ml-co2_data_ml['decimal date'])/co2_data_ml['sdev'])**2)
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
Input [0;32mIn [4][0m, in [0;36m<cell line: 29>[0;34m()[0m
[1;32m     23[0m [38;5;66;03m# Weighted Least Squares Mauna Loa [39;00m
[1;32m     24[0m p1_ml_fit, p1_ml_error [38;5;241m=[39m scipy[38;5;241m.[39moptimize[38;5;241m.[39mcurve_fit(p1,
[1;32m     25[0m     co2_data_ml[[38;5;124m'[39m[38;5;124mdecimal date[39m[38;5;124m'[39m],
[1;32m     26[0m     co2_data_ml[[38;5;124m'[39m[38;5;124maverage[39m[38;5;124m'[39m],
[1;32m     27[0m     sigma [38;5;241m=[39m co2_data_ml[[38;5;124m'[39m[38;5;124msdev[39m[38;5;124m'[39m]
[1;32m     28[0m     )
[0;32m---> 29[0m y_p1_ml [38;5;241m=[39m p3(co2_data_ml[[38;5;124m'[39m[38;5;124mdecimal date[39m[38;5;124m'[39m], [38;5;241m*[39m[43mp3_ml_fit[49m)
[1;32m     30[0m chisq_ml_p1 [38;5;241m=[39m np[38;5;241m.[39msum(((y_p1_ml[38;5;241m-[39mco2_data_ml[[38;5;124m'[39m[38;5;124mdecimal date[39m[38;5;124m'[39m])[38;5;241m/[39mco2_data_ml[[38;5;124m'[39m[38;5;124msdev[39m[38;5;124m'[39m])[38;5;241m*[39m[38;5;241m*[39m[38;5;241m2[39m)
[1;32m     32[0m p3_ml_fit, p3_ml_error [38;5;241m=[39m scipy[38;5;241m.[39moptimize[38;5;241m.[39mcurve_fit(p3,
[1;32m     33[0m     co2_data_ml[[38;5;124m'[39m[38;5;124mdecimal date[39m[38;5;124m'[39m],
[1;32m     34[0m     co2_data_ml[[38;5;124m'[39m[38;5;124maverage[39m[38;5;124m'[39m],
[1;32m     35[0m     sigma [38;5;241m=[39m co2_data_ml[[38;5;124m'[39m[38;5;124msdev[39m[38;5;124m'[39m]
[1;32m     36[0m     )

[0;31mNameError[0m: name 'p3_ml_fit' is not defined
NameError: name 'p3_ml_fit' is not defined

